tests/sycl/relational.cpp

```diff
@@ -0,0 +1,237 @@
+  // could also be done by using knowledge of the internal structure,
+  // but I wanted to avoid that.
+  template<typename T, std::enable_if_t<vector_length_v<T> == 0, int> = 0>
+  auto comp(T v, size_t idx) {
+    assert(idx == 0);
+    return v;
+  }
+  template<typename T, std::enable_if_t<vector_length_v<T> == 1, int> = 0>
+  auto comp(T v, size_t idx) {
+    assert(idx < vector_length_v<T>);
+    return v.x();
+  }
+  template<typename T, std::enable_if_t<vector_length_v<T> == 2, int> = 0>
+  auto comp(T v, size_t idx) {
+    assert(idx < vector_length_v<T>);
+    if(idx==0) return v.x();
+    return v.y();
+  }
+  template<typename T, std::enable_if_t<vector_length_v<T> == 3, int> = 0>
+  auto comp(T v, size_t idx) {
+    assert(idx < vector_length_v<T>);
+    if(idx==0) return v.x();
+    if(idx==1) return v.y();
+    return v.z();
+  }
+  template<typename T, std::enable_if_t<vector_length_v<T> == 4, int> = 0>
+  auto comp(T v, size_t idx) {
+    assert(idx < vector_length_v<T>);
+    if(idx==0) return v.x();
+    if(idx==1) return v.y();
+    if(idx==2) return v.z();
+    return v.w();
+  }
+  template<typename T, std::enable_if_t<vector_length_v<T> == 8, int> = 0>
+  auto comp(T v, size_t idx) {
+    assert(idx < vector_length_v<T>);
+    if(idx==0) return v.s0();
+    if(idx==1) return v.s1();
+    if(idx==2) return v.s2();
+    if(idx==3) return v.s3();
+    if(idx==4) return v.s4();
+    if(idx==5) return v.s5();
+    if(idx==6) return v.s6();
+    if(idx==7) return v.s7();
+    return v.s7();
+  }
+  template<typename T, std::enable_if_t<vector_length_v<T> == 16, int> = 0>
+  auto comp(T v, size_t idx) {
+    assert(idx < vector_length_v<T>);
+    if(idx==0) return v.s0();
+    if(idx==1) return v.s1();
+    if(idx==2) return v.s2();
+    if(idx==3) return v.s3();
+    if(idx==4) return v.s4();
+    if(idx==5) return v.s5();
+    if(idx==6) return v.s6();
+    if(idx==7) return v.s7();
+    if(idx==8) return v.s8();
+    if(idx==9) return v.s9();
+    if(idx==10) return v.sA();
+    if(idx==11) return v.sB();
+    if(idx==12) return v.sC();
+    if(idx==13) return v.sD();
+    if(idx==14) return v.sE();
+    if(idx==15) return v.sF();
+    return v.sF();
+  }
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(rel_genfloat_unary, T,
+                              rel_test_genfloats::type) {
+
+  constexpr int D = vector_length_v<T>;
+  using DT = vector_elem_t<T>;
+
+  namespace s = cl::sycl;
+
+  using IntType = typename std::conditional_t<std::is_same_v<DT, float>, int32_t, int64_t>;
+  using OutType = typename s::detail::builtin_type_traits<T>::template alternative_data_type<IntType>;
+
+  constexpr int FUN_COUNT = 5;
+
+  // build inputs and allocate outputs
+
+  s::queue queue;
+  s::buffer<T> in{{1}};
+  s::buffer<OutType> out{{FUN_COUNT}};
+  {
+    auto inputs  = in.template get_access<s::access::mode::write>();
+    auto outputs = out.template get_access<s::access::mode::write>();
```

Comment:
Better use `get_host_access()`, otherwise you get a deprecation warning once #979 is merged.

Reply:
Done.